kotlin/src/main/java/com/squareup/moshi/KotlinJsonAdapter.kt

```diff
@@ -143,61 +144,61 @@ internal class KotlinJsonAdapter<T> private constructor(
       return if (value !== ABSENT_VALUE) value else null
     }
   }
+}
 
-  companion object {
-    @JvmField val FACTORY = Factory { type, annotations, moshi ->
-      if (!annotations.isEmpty()) return@Factory null
+object KotlinJsonAdapterFactory : JsonAdapter.Factory {
```

Comment:
This has ugly implications for Java callers. We need `@file:JvmName` to get a decent named enclosing class and then we want this to be exposed ideally as a `FACTORY` field.